src/kernel/mem/pmm.c

```diff
@@ -75,6 +75,21 @@ void *pmm_alloc_frame(){
     return NULL;
 }
 
+void *pmm_alloc_area(size_t size) {
+    size_t requested_frames = get_number_of_pages_from_size(size);
+
+    loglinef(Verbose, "(pmm_alloc_area): requested_frames: %x\n", requested_frames);
+    uint64_t frames = _bitmap_request_frames(requested_frames);
```

Comment:
there is a potential race condition here between when you request the frames and set the bits.